File: src/main/java/waterpower/client/render/BlockColor.java
package waterpower.client.render;

import net.minecraft.block.state.IBlockState;
import net.minecraft.client.renderer.color.IBlockColor;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.IBlockAccess;
import waterpower.common.block.BlockWaterPower;

public class BlockColor implements IBlockColor {

	@Override
	public int colorMultiplier(IBlockState state, IBlockAccess worldIn, BlockPos pos, int tintIndex) {
		if (state.getBlock() instanceof BlockWaterPower) {
			return ((BlockWaterPower) state.getBlock()).colorMultiplier(state, worldIn, pos, tintIndex);
		} else return 0;
	}

}

File: src/main/kotlin/waterpower/client/render/block/BlockColor.kt
/**
 * Copyright (c) Huang Yuhui, 2017
 *
 * "WaterPower" is distributed under the terms of the Minecraft Mod Public
 * License 1.0, or MMPL. Please check the contents of the license located in
 * http://www.mod-buildcraft.com/MMPL-1.0.txt
 */
package waterpower.client.render.block

import net.minecraft.block.state.IBlockState
import net.minecraft.client.renderer.color.IBlockColor
import net.minecraft.util.math.BlockPos
import net.minecraft.world.IBlockAccess
import net.minecraftforge.fml.relauncher.Side
import net.minecraftforge.fml.relauncher.SideOnly
import waterpower.common.block.BlockBase

@SideOnly(Side.CLIENT)
object BlockColor : IBlockColor {
    override fun colorMultiplier(state: IBlockState, worldIn: IBlockAccess?, pos: BlockPos?, tintIndex: Int): Int {
        val block = state.block
        if (block is BlockBase) {
            return block.colorMultiplier(state, worldIn, pos, tintIndex)
        } else
            return 0
    }

}
